### PR TITLE
Add simple xlsx file render support

### LINKED ISSUE #4104: Default Excel/CSV files render supports
I propose to add render support for excel,csv files defaultly.

### PR DESCRIPTION
**Pull request by lunny**
partial fix #4104 

image

### CODE REVIEW

File: modules/markup/xlsx/xlsx.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
+// Copyright 20178 The Gitea Authors. All rights reserved.
```
Comment (axifive) [inline modules/markup/xlsx/xlsx.go:1]: typo, 2018

File: modules/markup/xlsx/xlsx.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	}
+
+	var tmpBlock bytes.Buffer
+	tmpBlock.WriteString(`<div class="ui top attached tabular menu">`)
+	for i := 0; i < len(sheetMap); i++ {
+		var active string
+		if i == 0 {
+			active = "active"
+		}
+		tmpBlock.WriteString(fmt.Sprintf(`<a class="%s item" data-tab="%d">%s</a>`, active, i, sheetMap[i+1]))
```
Comment (jvoisin) [inline modules/markup/xlsx/xlsx.go:53]: Shouldn't `sheetMap[i+1]` be escaped?

**Comment by lafriks** [comment]
It's actually quite small code on Gitea side so I would not mind it being included

**Comment by jvoisin** [comment]
Well, it's adding an external library as a dependency, increasing the attack surface of gitea by several thousands line of code.

**Comment by lafriks** [comment]
It should not be increasing attack vector as it just reads file. Stack/heap overflows should not be possible in Go and used lib is pure go code so it should be quite safe

**Comment by jvoisin** [comment]
It might add XSS, CSRF, arbitrary file inclusion, …

And yes, you can get [arbitrary remote code execution via go]( https://github.com/kacperszurek/exploits/blob/master/Gitea/gitea_lfs_rce.py ), even [in its stdlib]( https://groups.google.com/forum/#!topic/golang-announce/4rpTbfzYB1k ) :)

**Comment by lafriks** [comment]
There is risk but as we are using same sanitizer as markdown it should not be too high to not implement this feature

**Comment by jvoisin** [comment]
[Currently it's not]( https://github.com/go-gitea/gitea/pull/4106#pullrequestreview-126711936 ).

But to be honest, I'm more worried about adding an other repository for an attacker to take over to introduce malicious code into gitea. But I guess that it's how go is working :man_shrugging:

**Comment by lafriks** [comment]
@jvoisin I was talking about library not current PR code. What I was trying to say that library code should not be problem in this case, current PRs code of course needs to be fixed in multiple places.

**Comment by bkcsoft** [comment]
I'm with @jvoisin on this one... At least put this behind a build tag as it might introduce attack-vectors :neutral_face: (In before we need a build-service like Caddy :joy: )

**Comment by techknowlogick** [comment]
Closing due to external rendering available. Re-open if needed.

**Pull request closed without merging**